### PR TITLE
#162948855 bookmark 

### PR DESCRIPTION
**Pull request by kwameasiago**
#### What does this PR do?
Implement bookmarking featue

#### Description of Task to be completed?
created an endpoint for adding bookmarks
created an endpoint for removing bookmarks
created an endpoint for viewing bookmarks

#### How should this be manually tested?

- Clone the repo locally

- cd ah-django-unchained
- git checkout ft-bookmark-162948855
- use this [URL](http://127.0.0.1:8000/api/article/huge/[slug]/bookmark/) to add bookmarks
- use this [URL](http://127.0.0.1:8000/api/article/[slug]/delete/bookmark/) to delete bookmarks
- use this [URL](http://127.0.0.1:8000/api/article/bookmark/) to view all bookmarks

**Screenshots**
screenshot 2019-01-31 at 18 01 07
screenshot 2019-01-31 at 18 01 19

#### What are the relevant pivotal tracker stories?
[#162948855](https://www.pivotaltracker.com/story/show/162948855)

### CODE REVIEW

File: authors/apps/bookmark/models.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from django.db import models
+from ..articles.models import Article
+from ..authentication.models import User
+
+
+class Bookmarks(models.Model):
+    """
+    class for containg bookmarks model
```
Comment (MaggieChege) [inline authors/apps/bookmark/models.py]: typo in comment

Comment (MaggieChege) [reply authors/apps/bookmark/models.py]: Also, add a blank line at the end of the file.

File: authors/apps/bookmark/serializers.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from rest_framework import serializers
+from .models import Bookmarks
```
Comment (MaggieChege) [inline authors/apps/bookmark/serializers.py:2]: Has been imported twice

**Review by Georgeygigz — APPROVED** [review]

**Comment by Georgeygigz** [comment]
Kindly rebase your work

File: authors/apps/bookmark/views.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    queryset = Bookmarks.objects.all()
+    permission_classes = (IsAuthenticated,)
+
+    def list(self, request, *args, **kwargs):
+        bookmark = Bookmarks.objects.filter(user=request.user)
+        serializer = self.serializer_class(bookmark, many=True)
+        data = serializer.data
+        if len(data) == 0:
+            return Response(
+                {'error': 'You have no bookmark available'},
```
Comment (ignatiusukwuoma) [inline authors/apps/bookmark/views.py]: That a user doesn't have bookmarks is not really an error. Let's return an empty array instead

File: authors/apps/bookmark/urls.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from django.urls import path
+from .views import CreateBookmark, ListALlBookmarks, DeleteBookmark
+
+
+app_name = "bookmarks"
+urlpatterns = [
+    path('article/<slug>/create/bookmark/', CreateBookmark.as_view(), name='create-bookmark'),
+    path('article/<slug>/delete/bookmark/', DeleteBookmark.as_view(), name='delete-bookmark'),
```
Comment (ignatiusukwuoma) [inline authors/apps/bookmark/urls.py]: Since we are building a rest API, we should have one endpoint create and delete bookmarks

**Comment by ignatiusukwuoma** [comment]
I'm not sure why you decided to make `bookmarks` an app. it should have been built into articles since it's part of the article feature. You don't have to change it though, just implement the feedback.

**Review by ignatiusukwuoma — APPROVED** [review]

**Pull request merged**